__tests__/integration/editorTab.test.js

```diff
@@ -6,56 +6,42 @@ describe('Editor Tab', () => {
   })
 
   it('checks for headline', async () => {
-    const h3 = await page.$eval('#bfeditor-menudiv > h3', e => e.textContent)
-    await expect(h3).toMatch(/Create Resource/)
+    await expect_regex_in_selector_textContent('#bfeditor-menudiv > h3', /^Create Resource$/)
   })
 
   it('checks for default menu of profiles', async () => {
-    const monograph = await page.$eval('#bfeditor-menudiv > ul:nth-child(2) > li > a', e => e.textContent)
-    await expect(monograph).toMatch(/Monograph/)
-    const notated_music = await page.$eval('#bfeditor-menudiv > ul:nth-child(3) > li > a', e => e.textContent)
-    await expect(notated_music).toMatch(/Notated Music/)
-    const serial = await page.$eval('#bfeditor-menudiv > ul:nth-child(4) > li > a', e => e.textContent)
-    await expect(serial).toMatch(/Serial/)
-    const map = await page.$eval('#bfeditor-menudiv > ul:nth-child(5) > li > a', e => e.textContent)
-    await expect(map).toMatch(/Cartographic/)
-    const cd = await page.$eval('#bfeditor-menudiv > ul:nth-child(6) > li > a', e => e.textContent)
-    await expect(cd).toMatch(/Sound Recording: Audio CD/)
-    const cd_r = await page.$eval('#bfeditor-menudiv > ul:nth-child(7) > li > a', e => e.textContent)
-    await expect(cd_r).toMatch(/Sound Recording: Audio CD-R/)
-    const recording_analog = await page.$eval('#bfeditor-menudiv > ul:nth-child(8) > li > a', e => e.textContent)
-    await expect(recording_analog).toMatch(/Sound Recording: Analog/)
-    const blu_ray = await page.$eval('#bfeditor-menudiv > ul:nth-child(9) > li > a', e => e.textContent)
-    await expect(blu_ray).toMatch(/Moving Image: BluRay DVD/)
-    const moving_img_35 = await page.$eval('#bfeditor-menudiv > ul:nth-child(10) > li > a',  e => e.textContent)
-    await expect(moving_img_35).toMatch(/Moving Image: 35mm Feature Film/)
-    const print_photos = await page.$eval('#bfeditor-menudiv > ul:nth-child(11) > li > a', e => e.textContent)
-    await expect(print_photos).toMatch(/Prints and Photographs/)
-    const rare = await page.$eval('#bfeditor-menudiv > ul:nth-child(12) > li > a', e => e.textContent)
-    await expect(rare).toMatch(/Rare Materials/)
-    const authorities = await page.$eval('#bfeditor-menudiv > ul:nth-child(13) > li > a', e => e.textContent)
-    await expect(authorities).toMatch(/Authorities/)
+    await expect_regex_in_selector_textContent('#bfeditor-menudiv > ul:nth-child(2) > li > a', /^Monograph$/)
```

Comment:
I'm new to this particular tooling, but it seems like overkill to use regex matching for things that are just literal strings.  

I also don't understand why every criteria must be independently asynchronously paired (i.e. `await`ed). 

Is drilling down after a single `await` like this not viable?
```javascript
const menudiv = await(page.$eval('#bfeditor-menudiv'))
expect(menudiv.$eval('ul:nth-child(2) > li > a', e => e.textContent)).toBe('Monograph')
expect(menudiv.$eval('ul:nth-child(3) > li > a', e => e.textContent)).toBe('Notated Music')
```

Reply:
I am hearing two concerns in your comment:

1.  all the awaits.

$eval is a method on the page object via puppeteer's api (https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageevalselector-pagefunction-args-1 ), and it comprises a promise, which requires the await.  Presumably the promise/await mechanism of javascript is implemented such that the full page object is not reloaded.

Here's what I got when I tried your suggestion:

![image](https://user-images.githubusercontent.com/96775/47169494-8ca15980-d2b8-11e8-826c-491d3d4cf9e3.png)

2.  regex vs exact match

This is a good point.  I spent some time on this page https://jestjs.io/docs/en/expect and managed to miss "toBe".  I will update.

Reply:
In your example `menudiv` is actually a promise and not the element itself. If `await` is not used on the second line, the expect will fail immediately because the promise has not been fulfilled returning the element that is being tested.

Reply:
Thanks for the reply.  If I'm reading the docs correctly, would that be fixed by adding [`.resolves`](https://jestjs.io/docs/en/asynchronous#resolves-rejects)?

I get that I was treating return of `$eval` like it was jquery.  I guess if I wanted to do that, I would follow jest's ["DOM Manipulation" docs](https://jestjs.io/docs/en/tutorial-jquery) for actually having jquery.